test/DotNetty.Transport.Tests.Performance/Sockets/TcpSocketChannelPerfSpec.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace DotNetty.Transport.Tests.Performance.Sockets
+{
+    using System.Net;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using DotNetty.Buffers;
+    using DotNetty.Codecs;
+    using DotNetty.Transport.Bootstrapping;
+    using DotNetty.Transport.Channels;
+    using DotNetty.Transport.Channels.Sockets;
+    using DotNetty.Transport.Tests.Performance.Utilities;
+    using NBench;
+
+    public class TcpChannelPerfSpecs
+    {
+        const string InboundThroughputCounterName = "inbound ops";
+
+        const string OutboundThroughputCounterName = "outbound ops";
+
+        public const int WriteCount = 100000;
```

Comment:
Although, JIT technically should not be a factor - we run the entire spec 26 times, 2x the specified interval. The first 13 are warmups designed to JIT, align caches on individual cores, and so forth. But all of the math on those goes straight out the window as soon as the test has to run on more than one core like this one, because NBench doesn't set CPU affinity on concurrent tests (we specify that flag in FAKE.)